packages/fxa-settings/src/components/LinkValidator/index.tsx

```diff
@@ -28,6 +30,7 @@ const LinkValidator = <TModel extends ModelDataProvider>({
   linkType,
   viewName,
   getParamsFromModel,
+  integration,
```

Comment:
This is not directly related to this line, but close to it, and probably relevant to this PR.

I realized the getEmailsFromParams sort of by passes the model accessor pattern. I think we could clean this up if we did something like
```

// Just for illustration. This could be real model class, interface, or type. 
interface LinkValidtorModel {
    email:string|undefined
}

const LinkValidator = <TModel extends ModelDataProvider & LinkValidatorModel>({
  children,
  linkType,
  viewName,
  getParamsFromModel,
  integration,
}: LinkValidatorProps<TModel> & RouteComponentProps) => {
    const params = getParamsFromModel();
    // No more direct model access needed here now that we added 
    // the '& LinkValidatorModel' to the model type.
    const email = params.email;
    
```

Reply:
Seems fine to me. 👍 I left a comment about revisiting the LinkValidator pattern somewhere in this diff and refer to FXA-8099. Do you feel that and [this comment](https://github.com/mozilla/fxa/pull/15621#discussion_r1283671140) would be OK to address there, or would a separate issue be better?

Reply:
I'd be happy to address in FXA-8099. I'll make sure to ping you early and make sure we are on the same page for this change.